navigators/StackNavigator.js

```diff
@@ -0,0 +1,191 @@
+import { createNativeStackNavigator } from '@react-navigation/native-stack';
+import React from 'react';
+import GenreList from '../screens/common/GenreList';
+import CelebritiesHome from '../screens/home/CelebritiesHome';
+import MoviesHome from '../screens/home/MoviesHome';
+import TVShowsHome from '../screens/home/TVShowsHome';
+import MovieCollectionDetail from '../screens/movies/MovieCollectionDetail';
+import MovieCredits from '../screens/movies/MovieCredits';
+import MovieDetail from '../screens/movies/MovieDetail';
+import MovieList from '../screens/movies/MovieList';
+import PersonCreditsList from '../screens/person/PersonCreditsList';
+import PersonDetail from '../screens/person/PersonDetail';
+import PersonList from '../screens/person/PersonList';
+import TVSeasonDetail from '../screens/tv/TVSeasonDetail';
+import TvShowCredits from '../screens/tv/TvShowCredits';
+import TVShowDetail from '../screens/tv/TVShowDetail';
+import TVShowList from '../screens/tv/TVShowList';
+import { colorPrimaryDark } from '../utils/colors';
+import { ALL_SCREENS, CELEBRITIES_HOME, GENRE_LIST, MOVIES_HOME, MOVIE_COLLECTION_DETAIL, MOVIE_CREDITS, MOVIE_DETAIL, MOVIE_LIST, PERSON_CREDITS, PERSON_DETAIL, PERSON_LIST, TAB_STACK, TV_SEASON_DETAIL, TV_SHOWS_HOME, TV_SHOW_CREDITS, TV_SHOW_DETAIL, TV_SHOW_LIST } from './NavigatorNames';
+import TabNavigator from './TabNavigator';
+
+const Stack = createNativeStackNavigator();
+
+const screenOptions = {
+    headerTintColor: 'white',
+    headerStyle: {
+        backgroundColor: colorPrimaryDark
+    }
+};
+
+const MainStack = () => {
+    return (
+        <Stack.Navigator
+            screenOptions={{
+                headerShown: false
+            }}>
+            <Stack.Screen
+                name={TAB_STACK}
+                component={TabNavigator}
+            />
+            <Stack.Screen
+                name={ALL_SCREENS}
+                component={AllStack}
+            />
+        </Stack.Navigator>
+    );
+}
+
+const AllStack = () => {
+    return (
+        <Stack.Navigator
+            screenOptions={screenOptions}>
+            <Stack.Screen
+                name={MOVIE_DETAIL}
+                component={MovieDetail}
+                options={{
+                    title: ''
+                }}
+            />
+            <Stack.Screen
+                name={TV_SHOW_DETAIL}
+                component={TVShowDetail}
+                options={{
+                    title: ''
+                }} />
+            <Stack.Screen
+                name={PERSON_DETAIL}
+                component={PersonDetail}
+                options={{
+                    title: ''
+                }} />
+            <Stack.Screen
+                name={MOVIE_COLLECTION_DETAIL}
+                component={MovieCollectionDetail}
+                options={{
+                    title: ''
+                }} />
```

Comment:
Instead of writing title:' ' over and over, you can initialize it once 
screenTitle = {title:' '}
and use screenTitle everywhere. 

Also, check if it's possible to specify the same on navigator's screenOptions instead.